src/libraries/System.Runtime.Numerics/src/System/Numerics/BigNumber.cs

```diff
@@ -525,6 +559,231 @@ private static bool NumberToBigInteger(ref BigNumberBuffer number, out BigIntege
                     MultiplyAdd(ref currentBuffer, s_uint32PowersOfTen[partialDigitCount], partialValue);
                 }
 
+                result = NumberBufferToBigInteger(currentBuffer, number.sign);
+                return true;
+
+                bool ProcessChunk(ReadOnlySpan<char> chunkDigits, ref Span<uint> currentBuffer)
+                {
+                    int remainingIntDigitCount = Math.Max(numberScale - totalDigitCount, 0);
+                    ReadOnlySpan<char> intDigitsSpan = chunkDigits.Slice(0, Math.Min(remainingIntDigitCount, chunkDigits.Length));
+
+                    bool endReached = false;
+
+                    // Storing these captured variables in locals for faster access in the loop.
+                    uint _partialValue = partialValue;
+                    int _partialDigitCount = partialDigitCount;
+                    int _totalDigitCount = totalDigitCount;
+
```

Comment:
Naming locals with a `_` is not recommended these are reserved for private fields. Consider 
```cs
uint localPartialValue = partialValue;
```

Reply:
This part comes from the old source. I left it as is because I didn't think it should be changed carelessly. Of course, it is possible to change it. it's up to you.
https://github.com/dotnet/runtime/pull/55121/files/7f9fcd815e5efdd7a58703f2b53b611bd79a3b2e#diff-83081f6bd1123c4813c96a452ead0a065d670bbf44ed85023f772253eaf7d2c5L571-L603

Reply:
I'm fine with leaving the existing naming conventions as is for now. deeprobin is correct that it doesn't follow our current practice, but we can always clean it up as a separate PR.